--- glusterfs-8.2/xlators/features/bit-rot/src/bitd/bit-rot-scrub.c.orig	2020-09-29 08:27:05.478031023 -0400
+++ glusterfs-8.2/xlators/features/bit-rot/src/bitd/bit-rot-scrub.c	2020-09-29 08:28:35.165483744 -0400
@@ -1799,7 +1799,7 @@
     tmp_count = total_count;
 
     for (j = 0; j < count; j++) {
-        len = snprintf(key, PATH_MAX, "quarantine-%d", j);
+        len = snprintf(key, sizeof(key), "quarantine-%d", j);
         ret = dict_get_strn(child_dict, key, len, &entry);
         if (ret)
             continue;
@@ -1810,7 +1810,7 @@
         if ((len < 0) || (len >= PATH_MAX)) {
             continue;
         }
-        snprintf(main_key, PATH_MAX, "quarantine-%d", tmp_count);
+        snprintf(main_key, sizeof(main_key), "quarantine-%d", tmp_count);
 
         ret = dict_set_dynstr_with_alloc(dict, main_key, tmp);
         if (!ret)
