--- glusterfs-3.10.0rc1/api/src/glfs.c.orig	2017-02-22 12:36:41.962408227 -0500
+++ glusterfs-3.10.0rc1/api/src/glfs.c	2017-02-22 12:43:19.031223796 -0500
@@ -1471,6 +1471,8 @@
 {
         glusterfs_ctx_t  *ctx = NULL;
         int               ret = 0;
+        char              msg[1024] = {0,}; /* should not exceed 1024 chars */
+	size_t            rem = sizeof (msg);
 
         if (!fs || !fs->ctx) {
                 ret = -1;
@@ -1483,13 +1485,12 @@
         switch (sysrq) {
         case GLFS_SYSRQ_HELP:
         {
-                char msg[1024]; /* help text should not exceed 1024 chars */
                 struct glfs_sysrq_help *usage;
 
-                msg[0] = '\0';
                 for (usage = glfs_sysrq_help; usage->sysrq; usage++) {
-                        strncat (msg, usage->msg, 1024);
-                        strncat (msg, " ", 1024);
+                        strncat (msg, usage->msg, rem);
+			rem -= strlen (usage->msg);
+                        strncat (msg, " ", rem--);
                 }
 
                 /* not really an 'error', but make sure it gets logged */
